src/main/java/pl/gov/coi/pomocua/ads/iam/KeycloakUsersRepository.java

```diff
@@ -0,0 +1,26 @@
+package pl.gov.coi.pomocua.ads.iam;
+
+import lombok.RequiredArgsConstructor;
+import org.keycloak.admin.client.resource.UsersResource;
+import org.keycloak.representations.idm.UserRepresentation;
+import pl.gov.coi.pomocua.ads.UserId;
+import pl.gov.coi.pomocua.ads.users.User;
+import pl.gov.coi.pomocua.ads.users.UsersRepository;
+
+import javax.ws.rs.NotFoundException;
+
+@RequiredArgsConstructor
+public class KeycloakUsersRepository implements UsersRepository {
+
+    private final UsersResource usersResource;
+
+    @Override
+    public User getById(UserId userId) {
+        try {
+            UserRepresentation userRepresentation = usersResource.get(userId.value).toRepresentation();
+            return new User(userId, userRepresentation.getEmail());
+        } catch (NotFoundException e) {
+            throw new UserNotFoundException();
```

Comment:
I'm pretty sure that this would change the logic that's written in `UsersResource` - if no user is found there, this endpoint should return HTTP 401 (and there's logic for that there, but this exception will be thrown first).

Maybe we can follow the pattern from the `CurrentUser` interface, i.e. returning `Optional<User>` here and let the users of this interface decide what to do in case of the missing user?

Reply:
I'll change UsersRepository contract to return Optional<User>